app/assets/javascripts/angular/controllers/pages/create_recipe_controller.js.erb

```diff
@@ -1,43 +1,70 @@
 
-  var hourArray = [];
-  for (var i = 0; i < 24 ; i++) {
-    hourArray.push(i);
-  }
+  $scope.create_parameters = {};
 
-  var minuteArray = [];
-  for (var i = 0; i < 60 ; i++) {
-    minuteArray.push(i);
-  }
+  // var user = UserServices;
+  // $scope.create_parameters.author = UserServices.username;
 
-  $scope.hours = hourArray;
-  $scope.minutes = minuteArray;
+  $scope.create_parameters.title = '';
 
-  $scope.measurements = measures.sort()
-  // console.log($scope.measurements);
+  $scope.create_parameters.description = '';
 
-  var ingredients = [];
-  $scope.addIngredient = function() {
-    console.log($scope.addIngredNumber,$scope.addIngredUnit,$scope.addIngredName);
-
-    ingredient["number"] = $scope.addIngredNumber;
-    ingredient["unit"] = $scope.addIngredUnit;
-    ingredient["ingredient"] = $scope.addIngredName;
-    ingredients.push(ingredient);
-    $('.added-ingredients tbody').prepend("<tr class='added-ingredient'><td class='col-xs-2'></td><td class='col-xs-1'>-</td><td class='col-xs-1'>"+$scope.addIngredNumber+"</td><td class='col-xs-2'>"+$scope.addIngredUnit+"</td><td class='col-xs-6'>"+$scope.addIngredName+"</td></tr>");
-  }
+  { // add ingredients function
+    $scope.create_parameters.all_ingredients = [];
 
-  // time slider function
-  {
-    $scope.search_parameters.timeframe = [0,120];
+    var checkDuplicates = function (keyword,array) {
+      for (var i= 0 ; i < array.length ; i++) {
+        if (keyword == array[i]) {
+          return true;
+        }
+      }
+    };
+
+    $scope.newIngredient = function() {
+      if ($scope.addIngredName != undefined && $scope.addIngredName != "") {
+        if (checkDuplicates($scope.addIngredName,$scope.create_parameters.all_ingredients) == true) {
+          $scope.addIngredName = '';
+        } else {
+          $scope.create_parameters.all_ingredients.push({
+            "amount": $scope.addIngredNumber,
+            "unit": $scope.addIngredUnit,
+            "name": $scope.addIngredName
+          });
+          $scope.addIngredNumber = '';
+          $scope.addIngredUnit = '';
+          $scope.addIngredName = '';
+          // console.log($scope.addIngredName);
+          // console.log(checkDuplicates($scope.addIngredName,$scope.create_parameters.keywords));
+          console.log($scope.create_parameters);
+        }
+      } else {
+        $scope.addIngredName = '';
+        // needs notification of empty string
+      }
+      // console.log($scope.create_parameters.keywords);
+      // console.log($scope.create_parameters);
+    };
 
+    $scope.removeIngredient = function(index) {
+      $scope.create_parameters.all_ingredients.splice(index,1);
+    };
+  }
+  
+  { // time sliders function
```

Comment:
Try to remember to remove these  { } surrounding your code, it just helps you separate the different parts of your code, but if if doesn't break it doesn't mean that it makes sense